iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -307,6 +331,42 @@ public boolean supportsPartitionTransform() {
     }).collect(Collectors.toList());
   }
 
+  @Override
+  public DynamicPartitionCtx createDPContext(HiveConf hiveConf, org.apache.hadoop.hive.ql.metadata.Table hmsTable)
+      throws SemanticException {
+    TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
+    Table table = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
+    if (table.spec().isUnpartitioned()) {
+      return null;
+    }
+
+    // Iceberg currently doesn't have publicly accessible partition transform information, hence use above string parse
+    List<PartitionTransformSpec> partitionTransformSpecs = getPartitionTransformSpec(hmsTable);
```

Comment:
You might want beat me with a stick for mentioning this only now, but `getPartitionTransformSpec(hmsTable)` starts with:
```
    TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
    Table table = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
```
We have the Iceberg table at hand. Would it be better to have a `private partitionTransformSpec(Table iceTable)` which could be called here, and from `getPartitionTransformSpec`?

Reply:
As discussed offline this won't be addressed with this patch.